analyses/bfa/notebooks/bfa_seas_bavg_precip_corr_obsfor.md

```diff
@@ -527,47 +528,79 @@ If we want to continue understanding the suitability of this trigger, we therefo
 Better understand the historical trend and which years had extreme below average rainfall
 
 ```{code-cell} ipython3
-from scipy.interpolate import interp1d
+def get_return_periods(
+    df: pd.DataFrame,
+    rp_var: str,
+    years: list = None,
+    method: str = "analytical",
+    show_plots: bool = False,
+) -> pd.DataFrame:
+    """
+    :param df: Dataframe with data to compute rp on
+    :param rp_var: column name to compute return period on
+    :param years: Return period years to compute
+    :param method: Either "analytical" or "empirical"
+    :param show_plots: If method is analytical, can show the histogram and GEV distribution overlaid
+    :return: Dataframe with return period years as index and stations as columns
+    """
+    if years is None:
+        years = [1.5, 2, 3, 5]#, 10]#, 20]
+    df_rps = pd.DataFrame(columns=["rp"],index=years)
+    if method == "analytical":
+        f_rp = _get_return_period_function_analytical(
+            df_rp=df, rp_var=rp_var, show_plots=show_plots
+        )
+    elif method == "empirical":
+        f_rp = _get_return_period_function_empirical(
+            df_rp=df, rp_var=rp_var,
+        )
+    else:
+        logger.error(f"{method} is not a valid keyword for method")
+        return None
+    df_rps["rp"] = np.round(f_rp(years))
+    return df_rps
+```
+
+```{code-cell} ipython3
+years = np.arange(1.5, 9, 0.5)
+df_rps_analytical = get_return_periods(df_stats_reg, rp_var="bavg_cell",years=years, method="analytical", show_plots=True)
```

Comment:
Ah, do I understand correctly that values you are fitting the GEV to are percentages? In that case, the distribution has a known maximum (100%) and so the GEV wouldn't really work -- since it's a PDF, the integral should be 1. 

I know that this would mean not being able to compare regions, but did you think about using total area instead of percentage area (or even better, total area x population in that area?) Just because percentage area can lose a bit of meaning if the total area sizes are different and the population density is variable

Reply:
Very smart point :) 
It is only one region in this case. And the way it is computed it is hard to get the total area instead (or not worth the effort I would say). So will stick to empirical for now but this is good insight, thanks.